File: 04 - Tournament-Winner/04.py
def tournament_winner(competitions, results):
    winners = {}

    for i in range(len(results)):
        winner = competitions[i][1] if results[i] == 0 else competitions[i][0]
        if winner in winners.keys():
            winners[winner] += 3
        else:
            winners[winner] = 3

    finalWinner = max(winners, key=lambda k: winners[k])

    return finalWinner


print(
    "Solution:",
    tournament_winner(
        [
            ["HTML", "C#"],
            ["C#", "Python"],
            ["Python", "HTML"],
        ],
        [0, 0, 1],
    ),
)

File: easy/04 - Tournament-Winner/04.js
function tournamentWinner(competitions, results) {
    const winners = {}
  
    for (i=0; i<results.length; i++) {
      const winner = results[i] === 0 ? competitions[i][1] : competitions[i][0]
      if (winners.hasOwnProperty(winner)) {
        winners[winner] += 3
      }
      else {
        winners[winner] = 3
      }
    }
  
      const finalWinner = Object.keys(winners).reduce((a, b) => 
        winners[a] > winners[b] ? a : b);
  
      return finalWinner
}

console.log('Solution:', tournamentWinner(
       [
            ["HTML", "C#"],
            ["C#", "Python"],
            ["Python", "HTML"],
          ],
          [0, 0, 1]
      ));